testing/git_cached_repository/git_cached_repository.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+source ./testing/e2e/init.sh # Must be sourced 1st
+source ./testing/e2e/bazel.sh
+source ./testing/e2e/rules.sh
+source ./testing/e2e/unittest.sh
+source ./testing/e2e/logging.sh
+source ./testing/e2e/utils.sh
+source ./testing/e2e/repos.sh
+source ./testing/e2e/assert.sh
+source ./testing/e2e/testkit.sh
+
+# Must be invoked from the root of the repo.
+ROOT=$PWD
+
+function before_test() {
+  set_up
```

Comment:
idea - if `set_up` is loaded from `unittest.sh` then let's call it `unittest_set_up` - it'll help me in understanding the flow better.

Reply:
Renamed.